lib/console.js

```diff
@@ -111,47 +113,44 @@ function write(ignoreErrors, stream, string, errorhandler, groupIndent) {
 
     stream.write(string, errorhandler);
   } catch (e) {
+    if (MAX_STACK_MESSAGE === undefined) {
+      try {
+        // eslint-disable-next-line no-unused-vars
+        function a() { a(); }
+      } catch (err) {
+        MAX_STACK_MESSAGE = err.message;
+      }
+    }
     // console is a debugging utility, so it swallowing errors is not desirable
     // even in edge cases such as low stack space.
-    if (e.message === 'Maximum call stack size exceeded')
+    if (e.message === MAX_STACK_MESSAGE && e.name === 'RangeError')
       throw e;
     // Sorry, there’s no proper way to pass along the error here.
   } finally {
     stream.removeListener('error', noop);
   }
 }
 
-
-// As of v8 5.0.71.32, the combination of rest param, template string
-// and .apply(null, args) benchmarks consistently faster than using
-// the spread operator when calling util.format.
 Console.prototype.log = function log(...args) {
   write(this._ignoreErrors,
         this._stdout,
-        util.format.apply(null, args),
+        util.format(...args),
         this._stdoutErrorHandler,
         this[kGroupIndent]);
 };
-
-
 Console.prototype.debug = Console.prototype.log;
-
-
 Console.prototype.info = Console.prototype.log;
-
+Console.prototype.dirxml = Console.prototype.log;
 
 Console.prototype.warn = function warn(...args) {
   write(this._ignoreErrors,
         this._stderr,
-        util.format.apply(null, args),
+        util.format(...args),
```

Comment:
The spread operator, unlike .apply(), pushes the elements onto the stack.  That is, it makes stack overflows more likely.